Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTBaseClient.java

```diff
@@ -70,6 +79,20 @@ public Object invokePort(Class<T> portClass, String methodName, Object... operat
         serviceEndpoint = new URLServiceEndpointDecorator<>(serviceEndpoint, url);
         serviceEndpoint = new TimeoutServiceEndpointDecorator<>(serviceEndpoint, timeout);
 
+        boolean isDisableCNCheck = false;
+        try {
+            isDisableCNCheck = PropertyAccessor.getInstance().getPropertyBoolean(GATEWAY_PROPERTY_FILE,
+                DISABLE_CN_CHECK);
+        } catch (PropertyAccessException ex) {
+            LOG.error("Not able to 'disableCNCheck' from the property file: {}", ex.getLocalizedMessage(), ex);
+        }
+
+        LOG.info("gateway-property--disableCNCheck: '{}'", isDisableCNCheck);
+
+        if (isDisableCNCheck) {
```

Comment:
Since ConnectSecureClient use TLSClientServiceEndpointDecorator inside CachingCXFSecuredServicePortBuilder, CN check still doesn't apply when I need to turn on/off for secure.  Only add TLSClientServiceEndpointDecorator here.  Move yr check into TLSClientParametersFactory.  Also, verify if we need to remove TLSClientServiceEndpointDecorator from CachingCXFSecuredServicePortBuilder?